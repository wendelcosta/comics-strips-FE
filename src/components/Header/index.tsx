import React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) =>
  createStyles({
    header: {
      textAlign: 'center',
      backgroundColor: theme.custom.blue,
      color: theme.custom.white,
      '& h1': {
        marginTop: 0,
        padding: 10,
      },
    },
  })
)

const Header: React.FC = () => {
  const classes = useStyles()
  return (
    <header className={classes.header}>
      <h1>Comics xkcd - Brought to you by Wendel Costa</h1>
    </header>
  )
}

export default Header
