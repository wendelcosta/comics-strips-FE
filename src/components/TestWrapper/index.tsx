import React from 'react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../../styles/theme'

const TestWrapper: React.FC = ({ children }: any) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default TestWrapper
