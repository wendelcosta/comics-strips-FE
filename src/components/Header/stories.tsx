import React from 'react'

import Header from '.'
import TestWrapper from '../TestWrapper'

export default {
  title: 'Header',
  component: Header,
  argTypes: {
    text: { control: 'text' },
  },
}

export const Default = (args: Record<string, string>): any => (
  <TestWrapper>
    <Header text={args.text} {...args} />
  </TestWrapper>
)
Default.args = {
  text: 'Comics xkcd - Brought to you by Wendel Costa',
}
