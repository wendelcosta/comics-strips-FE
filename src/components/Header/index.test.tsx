import React from 'react'
import { render } from '@testing-library/react'

import TestWrapper from '../TestWrapper'

import Header from '.'

describe('Header Test', () => {
  test('Loads Header', () => {
    const { container } = render(
      <TestWrapper>
        <Header />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
