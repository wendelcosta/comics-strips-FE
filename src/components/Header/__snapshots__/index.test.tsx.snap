// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Loads Header 1`] = `
<div>
  <header
    class="makeStyles-header-1"
  >
    <h1>
      Comics xkcd - Brought to you by Wendel Costa
    </h1>
  </header>
</div>
`;
